src/olympia/reviewers/utils.py

```diff
@@ -1360,6 +1373,55 @@ def reject_multiple_versions(self):
                 content_review=self.content_review,
             )
 
+    def unreject_multiple_versions(self):
+        """Un-reject a list of versions."""
+        # self.version and self.file won't point to the versions we want to
+        # modify in this action, so set them to None before finding the right
+        # versions.
+        latest_version = self.version
+        self.version = None
+        self.file = None
+        now = datetime.now()
+        # we're only supporting non-automated reviews right now:
+        assert self.human_review
+
+        log.info(
+            'Making %s versions %s awaiting review (not disabled)'
+            % (self.addon, ', '.join(str(v.pk) for v in self.data['versions']))
+        )
+
+        for version in self.data['versions']:
+            self.set_file(amo.STATUS_AWAITING_REVIEW, version.file)
+
+            self.log_action(
+                action=amo.LOG.UNREJECT_VERSION,
+                version=version,
+                file=version.file,
+                timestamp=now,
+                user=self.user,
+            )
+
+        # if these are listed versions then the addon status may need updating
+        if (
+            self.data['versions']
+            and latest_version.channel == amo.CHANNEL_LISTED
+            and self.addon.status == amo.STATUS_NULL
+        ):
+            log.info(
+                'Changing add-on status [%s]: %s => %s (%s).'
+                % (
+                    self.addon.id,
+                    self.addon.status,
+                    amo.STATUS_NOMINATED,
+                    'unrejecting versions',
+                )
+            )
+            self.addon.update(status=amo.STATUS_NOMINATED)
+            ActivityLog.create(
+                amo.LOG.CHANGE_STATUS, self.addon, self.addon.status, user=self.user
+            )
+            self.addon.update_version()
```

Comment:
I don't love doing all this manually... Why `STATUS_NOMINATED` in particular and not `STATUS_APPROVED` ? Edge cases around add-on not actually ready to be public ?

Reply:
We don't have a method in the model that supports status being recalculated "up" currently (from `STATUS_NULL` to a higher status).  We could move this functionality into a new method on `Addon` if you prefer?  (it'd still be the same "manual" code though).

> Why STATUS_NOMINATED in particular and not STATUS_APPROVED ? Edge cases around add-on not actually ready to be public ?

The versions are changed to `STATUS_AWAITING_REVIEW`, so `STATUS_NOMINATED` is the correct status when there is a listed version with that status but no approved versions. (If you're asking why we aren't changing the versions to `STATUS_APPROVED`: the ask, as I understood it, was for an action to change the versions back to a pending state so they could then be manually approved by the reviewer - otherwise the request would have been to have allowed the existing approve action to be applied to `STATUS_DISABLED` versions instead)

Reply:
Do you think we could roll that into `update_status()` ? Wondering if there is any reason to keep updating the add-on status manually in reviewer tools/devhub/API these days. It should be possible to determine it purely by looking at the attached versions right ?

Reply:
I think the reason historically was were were afraid of weird file statuses coming from historic bad code, changes to what statuses mean, or just incomplete manual admin changes, so we didn't want an add-on that was status incomplete to "magically" change to "nominated" (or worse - change to "approved").  

Even today I'm cautious of doing it automatically in `update_status` - an add-on is only set to `STATUS_NOMINATED` in devhub and the api if all it's metadata is set - it's not just based on the file status (in the reviewer tools we're making assumptions that the action would only be used on add-ons that were previously okay to be in the `STATUS_NOMINATED` state and/or the reviewer knows what they're doing).  And we have a very limited number of places where an add-on could possibly change status to `STATUS_NOMINATED` (rather than every single time we call `update_status` - a lot)

That said, we could.  Options:
- another function like `update_status` that just handles this case
- an optional kwarg to `update_status` that unlocks an "upgrade" from `STATUS_NULL` to `STATUS_NOMINATED` if appropriate.
- no kwarg but an additional check on the metadata being valid too.

Reply:
I think even in reviewer tools, it shouldn't be possible to update the status to `STATUS_NOMINATED` (or `STATUS_APPROVED`...) when required metadata is absent. We should fail loudly instead... So I would vote for no kwarg...

But it should probably be done separately, as changing `update_status()` has the potential to reveal more edge cases unrelated to this feature. So in the meantime, probably just extract that code to some method that we would eventually refactor away, and file a follow-up issue to clean everything up ?

Reply:
https://github.com/mozilla/addons-server/pull/19728/commits/4b8bfdc64ee95100d6e5865033f18492adf84066 moves the nominated status setting up to `Addon` model.  I'll file an issue to refactor it into `update_status` (in some way)

Reply:
https://github.com/mozilla/addons-server/issues/19773